hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/OzoneServiceProvider.java

```diff
@@ -49,34 +49,24 @@
 
   @PostConstruct
   public void init() {
-    Collection<String> serviceIdList =
-        conf.getTrimmedStringCollection(OZONE_OM_SERVICE_IDS_KEY);
-    if (serviceIdList.size() == 0) {
-      // Non-HA cluster
+    try {
+      omserviceID = OmUtils.getOzoneManagerServiceId(conf);
+    } catch (IOException ex) {
+      throw new ConfigurationException(ex.getMessage(), ex);
+    }
+
+    // Non-HA cluster.
+    if (omserviceID == null) {
       omServiceAddr = SecurityUtil.buildTokenService(OmUtils.
           getOmAddressForClients(conf));
     } else {
-      // HA cluster.
-      //For now if multiple service id's are configured we throw exception.
-      // As if multiple service id's are configured, S3Gateway will not be
-      // knowing which one to talk to. In future, if OM federation is supported
-      // we can resolve this by having another property like
-      // ozone.om.internal.service.id.
-      // TODO: Revisit this later.
-      if (serviceIdList.size() > 1) {
-        throw new IllegalArgumentException("Multiple serviceIds are " +
-            "configured. " + Arrays.toString(serviceIdList.toArray()));
-      } else {
-        String serviceId = serviceIdList.iterator().next();
-        Collection<String> omNodeIds = OmUtils.getOMNodeIds(conf, serviceId);
-        if (omNodeIds.size() == 0) {
-          throw new IllegalArgumentException(OZONE_OM_NODES_KEY
-              + "." + serviceId + " is not defined");
-        }
-        omServiceAddr = new Text(OzoneS3Util.buildServiceNameForToken(conf,
-            serviceId, omNodeIds));
-        omserviceID = serviceId;
+      Collection<String> omNodeIds = OmUtils.getOMNodeIds(conf, omserviceID);
```

Comment:
existing om-ha-s3g or om-ha docker-compose don't have multiple service id configured. Can we add a unit or acceptance test to validate the behavior?